metrics.py

```diff
@@ -0,0 +1,184 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+r"""Used to collect anonymous DeepVariant metrics."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import atexit
+import json
+import logging
+import time
+import uuid
+import requests
+
+_CLEARCUT_ENDPOINT = 'https://play.googleapis.com/log'
+_CLOUD_HCLS = 'CLOUD_HCLS'
+_CONCORD = 'CONCORD'
+_DEEP_VARIANT_RUN = 'DeepVariantRun'
+_HTTP_REQUEST_TIMEOUT_SEC = 10
+_PYTHON = 'PYTHON'
+_VIRTUAL_CHC_DEEPVARIANT = 'virtual.chc.deepvariant'
+
+
+def capture_exceptions(func):
+  """Function decorator to capture and log any exceptions."""
+
+  def wrapper(*args, **kwds):
+    try:
+      return func(*args, **kwds)
+    # pylint:disable=broad-except
+    except Exception as e:
+      logging.error('Exception captured in %s : %s', func.__name__, e)
+
+  return wrapper
+
+
+class _ConcordEvent(object):
+  """Encapsulates information representing a Concord event."""
+
+  def __init__(self,
+               event_name,
+               event_type,
+               project_number,
+               console_type,
+               page_hostname,
+               event_metadata=None):
+    self._event_name = event_name
+    self._event_type = event_type
+    self._project_number = project_number
+    self._console_type = console_type
+    self._page_hostname = page_hostname
+    self._event_metadata = event_metadata or {}
+
+  def to_json(self, **kwargs):
+    """Encodes data in json."""
+    event_dict = {
+        'project_number': self._project_number,
+        'event_name': self._event_name,
+        'event_type': self._event_type,
+        'console_type': self._console_type,
+        'page_hostname': self._page_hostname,
+        'event_metadata': self._event_metadata_as_kv(),
+    }
+    return json.dumps(event_dict, **kwargs)
+
+  def _event_metadata_as_kv(self):
+    # pylint:disable=g-complex-comprehension
+    return [{
+        'key': k,
+        'value': str(v)
+    } for k, v in sorted(self._event_metadata.items())]
+
+
+class _MetricsCollector(object):
+  """Singleton class that collects and submits metrics.
+
+    Instances of this class share the same internal state, and thus behave
+    the same all the time.
+  """
+  _shared_events = []
+  _shared_session_identifier = uuid.uuid4().hex
+
+  def __init__(self):
+    self._events = self._shared_events
```

Comment:
I don't think you need to define self._events or self._session_identifier, you can just access `_shared_events` and ` _shared_session_identifier` as the static variables.